build/Mlos.Common.targets

```diff
@@ -4,4 +4,7 @@
     <MSBuild Projects="@(ProjectReference)" Targets="Restore" RunEachTargetSeparately="true" />
     <MSBuild Projects="@(ProjectFile)" Targets="Restore" RunEachTargetSeparately="true" />
   </Target>
+  <!-- An empty placeholder for other targets to reference with an AfterTargets property -->
+  <!-- We can then invoke this one via "dotnet build /t:PublishProjectBinary" -->
+  <Target Name="PublishProjectBinary" />
```

Comment:
Publish is for NetCore, why we need it in common ?

Reply:
That's a no-op definition.  It allows calling the target from a `dirs.proj` to publish recursively.

Reply:
I know it is no op. 
I do not understand why do we need to publish host depended binaries in the build system.
We had this before , and I had it removed to reduce nuget footprint.

Reply:
What we need to have is the settings registry in one folder, right ?
Maybe let's just modify the binplacing target (or a new one)
Mlos.SettingsSystem.CodeGen.targets 
  <!-- Binplace settings registry to the target folder -->
  <Target Name="BinplaceSettingsRegistry" AfterTargets="Build">
    <Copy SourceFiles="$(OutputPath)\$(TargetFileName)" DestinationFolder="$(BinplaceDir)" />
  </Target>
?

Reply:
I didn't realize those were included in the nuget.  Seems like that should probably be more specialized as well, no?

Anyways, as I mentioned elsewhere, I thought about this somemore and agree - publishing is unnnecessary.  I will yank that out in favor of binplacing the AnyCPU instead and calling it with `dotnet`.

Modifying the `Mlos.SettingsSystem.CodeGen.targets` to place everything in a single folder only maybe helps us for things internal to the MLOS repo.

Once we start add external project tuning (later this week?), then we'll still need to support some sort of search path since different projects will need different output locations.

Reply:
Reverted and turned them into binplace operations instead.